src/raft_log.rs

```diff
@@ -481,6 +499,38 @@ impl<T: Storage> RaftLog<T> {
         self.committed = snapshot.get_metadata().get_index();
         self.unstable.restore(snapshot);
     }
+
+    /// increase_uncommitted_size computes the size of the proposed entries and
+    /// determines whether they would push leader over its max_uncommitted_entries_size limit.
+    /// If the new entries would exceed the limit, the method returns false. If not,
+    /// the increase in uncommitted entry size is recorded and the method returns
+    /// true.
```

Comment:
This needs a `# Panics` section which details that passing `0` as `max` will cause a panic.

Reply:
I only added the panic to help me chase down an annoying bug where I broke all the tests. Can just pull it out.

Reply:
Ok. :)